android/app/src/main/java/com/chorestory/fragment/ParentQuestsFragment.java

```diff
@@ -1,47 +1,125 @@
+import com.chorestory.model.QuestParcelable;
+import com.chorestory.templates.ClanChildrenResponse;
+import com.chorestory.templates.ClanResponse;
+import com.chorestory.templates.GetQuestsResponse;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
 public class ParentQuestsFragment extends ChoreStoryFragment {
 
+    @Inject
+    RetrofitInterface retrofitInference;
+    @Inject
+    TokenHandler tokenHandler;
+
+    private long SECONDS_IN_WEEK = 604800;
+
     private ViewPager viewPager;
     private TabLayout tabLayout;
 
+    private ArrayList<QuestParcelable> questParcelables;
+
     private QuestsAdapter questsAdapter;
 
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_parent_quests, container, false);
 
+        App.getAppComponent().inject(this);
+
         viewPager = view.findViewById(R.id.view_pager);
-        setupViewPager();
         tabLayout = view.findViewById(R.id.tab_layout);
-        tabLayout.setupWithViewPager(viewPager);
         tabLayout.setSelectedTabIndicatorHeight(0);
-
         return view;
     }
 
     private void setupViewPager() {
-        List<Fragment> fragmentList = Arrays.asList(new QuestsUpcomingFragment(),
-                new QuestsPendingFragment(),
-                new QuestsCompletedFragment());
+        Bundle bundle = new Bundle();
+        bundle.putParcelableArrayList("all_quests", questParcelables);
+
+        QuestsUpcomingFragment questsUpcomingFragment = new QuestsUpcomingFragment();
+        questsUpcomingFragment.setArguments(bundle);
+        QuestsPendingFragment questsPendingFragment = new QuestsPendingFragment();
+        questsPendingFragment.setArguments(bundle);
+        QuestsCompletedFragment questsCompletedFragment = new QuestsCompletedFragment();
+        questsCompletedFragment.setArguments(bundle);
+
+        List<Fragment> fragmentList = Arrays.asList(questsUpcomingFragment,
+                questsPendingFragment,
+                questsCompletedFragment);
         List<String> fragmentTitleList = Arrays.asList(getString(R.string.upcoming),
                 getString(R.string.pending),
                 getString(R.string.completed));
         questsAdapter = new QuestsAdapter(getChildFragmentManager(), fragmentList, fragmentTitleList);
         viewPager.setAdapter(questsAdapter);
     }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        getQuests();
+    }
+
+    private void getQuests() {
+        String token = tokenHandler.getToken(getContext());
+        if (token != null) {
+            if (tokenHandler.isParentToken(token)) {
+                long currentTime = System.currentTimeMillis()/1000;
+                long start = currentTime - SECONDS_IN_WEEK;
```

Comment:
I thought we were not doing 2 weeks

Reply:
I'm still going to do 2 weeks but if it's a recurring quest then I'll only use one occurrence, so there's no issues